disttask/framework/storage/task_table.go

```diff
@@ -334,10 +334,24 @@ func (stm *TaskManager) GetSubtaskInStates(tidbID string, taskID int64, step int
 	if err != nil {
 		return nil, err
 	}
-	if len(rs) == 0 {
+
+	subtasks := make([]*proto.Subtask, len(rs))
+	for i, row := range rs {
+		subtasks[i] = row2SubTask(row)
+	}
+	return subtasks, nil
+}
+
+// GetFirstSubtaskInStates gets the first subtask by given states.
+func (stm *TaskManager) GetFirstSubtaskInStates(tidbID string, taskID int64, step int64, states ...interface{}) (*proto.Subtask, error) {
+	subtasks, err := stm.GetSubtasksInStates(tidbID, taskID, step, states...)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
If this is implemented by calling another method of the same interface, I think there is no need to provide two different methods...

How about adding the clause `LIMIT 1` in SQL as an optimization?